src/plugins/intel_cpu/src/transformations/cpu_opset/x64/pass/sdpa_fuse_transpose_reshape.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "sdpa_fuse_transpose_reshape.hpp"
+
+#include <transformations/utils/utils.hpp>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/scaled_dot_product_attention.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/cpu_opset/common/op/sdpa.hpp"
+
+/*
+ * Description: SDPA fuse transpose and reshape.
+ *           Original pattern                            Fused pattern
+ *
+ *  input1         input2       input3
+ *     |             |             |
+ * q_reshape     k_reshape     v_reshap
```

Comment:
Yes, `Reshape` is `inplace` in this case. 
`Transpose` take more time, because it needs to  copy big data.
Although `Reshape` takes almost no time, it is still called during execution, and the function call itself also takes time, (although very little).

In this case, since `Reshape` is also easy to fuse, so I do it. @dmitry-gorokhov